### PR TITLE
Build doc on RTD with pip (drop conda)

### PR DESCRIPTION
**Pull request by saimn**
Ref #699

For reference, on my fork:
- https://readthedocs.org/projects/photutils-test/builds/7580522/ ( 831 sec.)
- https://readthedocs.org/projects/photutils-test/builds/7580641/ ( 895 sec.)

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -32,12 +32,15 @@
...
-except ImportError:
+    astropy_helpers.sphinx
+except (ImportError, AttributeError):
     # Building from inside the docs/ directory?
     if os.path.basename(os.getcwd()) == 'docs':
         a_h_path = os.path.abspath(os.path.join('..', 'astropy_helpers'))
         if os.path.isdir(a_h_path):
             sys.path.insert(1, a_h_path)
+        if 'astropy_helpers' in sys.modules:
+            del sys.modules['astropy_helpers']
```
Comment (saimn) [inline docs/conf.py]: This is to fix astropy_helpers import on Python 3, when not using `setup.py build_docs` (RTD uses sphinx-build directly).

**Comment by larrybradley** [comment]
Many thanks, @saimn!

**Comment by saimn** [comment]
@larrybradley - I guess the "use system site-packages" box is checked in "advanced settings" in RTD's settings. That would explain why the colormap changed to jet, as it means that an old version of matplotlib from the system is used. But this also has the benefit of building more quickly!

**Comment by larrybradley** [comment]
Yes, the "Give the virtual environment access to the global site-packages dir" box is checked.  I wondered why the colormap was jet.  I explicitly set it to viridis in https://github.com/astropy/photutils/commit/82f0493c2338bf0880c763592dd46b98175c68df.

**Comment by bsipocz** [comment]
Wow @saimn, you're a saviour. The latest build is down to 470s.

**Comment by larrybradley** [comment]
I wonder what version of `matplotlib` is in the RTD system.  I also just noticed that the default `imshow` interpolation isn't `'nearest'`, so that will have to explicitly be set too. ☹️

**Comment by larrybradley** [comment]
Should I try *without* using the system site-packages??

**Comment by bsipocz** [comment]
offtopic here, but I wonder whether @saimn you want to experiment with the same thing for astropy? We see these timeouts there, too.

**Comment by saimn** [comment]
> Should I try without using the system site-packages??

Yes, I think it's better on the long-term to isolate from the system packages. Here it probably builds on an old Ubuntu LTS, so a matplotlib version < 2.0 (without the new defaults, style, colormap, and options like interpolation).

> offtopic here, but I wonder whether @saimn you want to experiment with the same thing for astropy?

I can have a try yes, out of curiosity.

**Comment by larrybradley** [comment]
@saimn I agree.  After unsuccessfully trying to update matplotlib to >= 2.0, I turned off the system package last night.

**Pull request merged**